android/capacitor/src/main/java/com/getcapacitor/Bridge.java

```diff
@@ -194,10 +211,70 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
         }
       }
     });
+  }
 
+  private void loadCustomWebView() {
+    webView.setWebChromeClient(new BridgeWebChromeClient(this));
 
-    // Get to work
-    webView.loadUrl(appUrl);
+    // Saving off old client so no functionality is lost if
+    // custom webview implements an existing client
+    final WebViewClient oldClient = webView.getWebViewClient();
+    webView.setWebViewClient(new WebViewClient() {
+
+      @Override
+      public void onPageStarted(WebView view, String url, Bitmap favicon) {
+        if (oldClient != null) {
+          oldClient.onPageStarted(view, url, favicon);
+        }
+      }
+
+      @Override
+      public void onPageFinished(WebView view, final String location) {
+        if (oldClient != null) {
+          oldClient.onPageFinished(view, location);
+        }
+
+        JSInjector injector = getJSInjector();
+        injectScriptFile(view, injector.getScriptString());
+      }
+
+      private void injectScriptFile(WebView view, String script) {
+
+        // Base64 encode string before injecting as innerHTML
```

Comment:
What's the encoding process for?

Reply:
Since I'm using the `innerHTML` here, and writing javascript from java, I want to make sure that nothing with quotes or double quotes gets messed up when writing/injects of this javascript. So I base64 encode the string in java and then base64 decode the string using `window.atob` for safety